shared/logic/IntraTickCache.ts

```diff
@@ -51,6 +51,18 @@ export function getBuildingIO(
             resources[k] = 1;
          });
       }
+      if ("sellResources" in b && type === "output") {
+         const capacity = b.capacity * b.level * totalMultiplierFor(xy, "output", 1, gs);
+         forEach((b as IMarketBuildingData).sellResources, (k) => {
+            const buyResource = (b as IMarketBuildingData).availableResources[k]!;
+            result[buyResource] = round(
```

Comment:
The logic here feels like it must be implemented somewhere - we should probably resue that. (if there's no existing function, maybe create one)

Also usually we do not round in the logic (rounding is only done in the view layer)

Reply:
It was also in MarketBuildingBody.  Have move the code to a function in BuildingLOgic.ts